apps/api/src/middleware/rate-limiter.middleware.ts

```diff
@@ -77,12 +126,38 @@ export const rateLimiterMiddleware = async (
       return next();
     }
 
-    // Get agent ID from request (set by auth middleware)
-    const agentId = req.agentId || "anonymous";
+    // Get agent ID from request (set by auth middleware) or use IP address as fallback
+    let agentId = req.agentId || `ip:${getClientIp(req)}`;
```

Comment:
Just noticed this line. That's great! It already implements the idea to use the IP only as a fallback. 👍